### ISSUE TITLE
Web Extensions: duplicate bundles when importing the same file via `import` and `manifest.json`

### ISSUE THREAD

**Issue by fregante** [parcel-bundler/parcel#9446]
# 🐛 bug report

In my extension, I mention the same file in the manifest and in a JS file. This causes Parcel to bundle the same file twice with two outputs. This in turn causes nearly every sub-import to also be duplicated. In this case the duplicate file was `source/index.html` (note that this screenshot was made after `trash distribution && parcel build`)

<img width="439" alt="Screenshot" src="https://github.com/parcel-bundler/parcel/assets/1402241/283933b1-34de-4690-a66e-99e51a8aaba4">

## 🎛 Configuration (.babelrc, package.json, cli command)

```js
{
	"extends": "@parcel/config-webextension",
	"transformers": {
		"*.svelte": ["parcel-transformer-svelte"]
	}
}
```

```js
{
	"@parcel/resolver-default": {
		"packageExports": true
	}
}
```

## 🤔 Expected Behavior

The file should not be duplicated.

## 😯 Current Behavior

Two `html` files appear in `dist`

## 💻 Code Sample

```json
{
	"name": "name",
	"description": "desc",
	"version": "0.0.0",
	"manifest_version": 3,
	"action": {
		"default_popup": "index.html"
	}
	"background": {
		"service_worker": "background.js",
		"type": "module"
	},
}
```

```js
// background.js

import appUrl from 'url:./index.html';
console.log(appUrl);
```

Full repro can be found in this PR (until commit `1009154`)

- https://github.com/hankxdev/one-click-extensions-manager/pull/125

Note that in my case, the manifest actually mentions `index.html?type=popup`, which is valid and could further cause issues, but the same bug appears even without the search parameter.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           |2.10.3|
| Node             | v18.18.2
| npm/Yarn         |9.8.1
| Operating System |macOS

**Comment by fregante** [comment]
I also tried using the same exact path in both, to no avail:

```js
import appUrl from 'url:./index.html';
```

```json
	"action": {
		"default_icon": "logo.png",
		"default_popup": "./index.html"
	},
```

**Comment by mischnic** [comment]
Does `new URL("index.html", import.meta.url)` cause the same problem?

**Comment by fregante** [comment]
Yes. I also tried matching the exact string in both ways: `index.html`, `./index.html`

**Comment by fregante** [comment]
Ping

**Comment by fregante** [comment]
Still reproducible with today's canary. Sub-dependencies are also duplicated:

- `file.hash.html` imports:
	- `file.hash.css`
	- `file.hash.js`
- `file.anotherhash.html` imports:
	- `file.anotherhash.css`
	- `file.anotherhash.js`

contents are 100% match except these import paths.

**Comment by fregante** [comment]
I'm seeing the same issue with just `manifest.json` in the latest canary after this was fixed:

- https://github.com/parcel-bundler/parcel/issues/9458

The repro:

```json
{
	"background": {
		"type": "module"
		"service_worker": "background.js",
		"scripts": [
			"background.js"
		]
	},
}
```

The generated `background.29017fde.js` and `background.e1bfa0d5.js` are identical except the internal module IDs

**Comment by fregante** [comment]
.
